app/core/llm_framework/openai_vanilla.py

```diff
@@ -16,6 +16,8 @@ def get_context(source_documents):
     context = '['
     # ** This will need to be adjusted, based on what the returned results look like **
     for i in range(len(source_documents)):
+        if len(source_documents[i].page_content) + len(context) > 14000:  # Maybe 3,500 tokens? Ideally we would estimate this based on tokens
```

Comment:
We can use the tiktoken library to count tokens. Later though. I'll suggest adding a FIXME for now so we can find it later. 

```suggestion
        if len(source_documents[i].page_content) + len(context) > 14000:  # FIXME: use tiktoken library to count tokens
```